notes_app/notes/api.py

```diff
@@ -39,6 +39,44 @@ def delete_detail(self, object_list, bundle):
         raise Unauthorized("Deletes not allowed")
 
 
+# Custom User Authorization
+class UserObjectsOnlyAuthorization(Authorization):
+    def read_list(self, object_list, bundle):
+        # This assumes a ``QuerySet`` from ``ModelResource``.
+        return object_list.filter(user=bundle.request.user)
+
+    def read_detail(self, object_list, bundle):
+        # Is the requested object owned by the user?
+        return bundle.obj.user == bundle.request.user
+
+    def create_list(self, object_list, bundle):
+        # Assuming they're auto-assigned to ``user``.
+        return object_list
+
+    def create_detail(self, object_list, bundle):
+        return bundle.obj.user == bundle.request.user
+
+    def update_list(self, object_list, bundle):
+        allowed = []
```

Comment:
this is the worst way for working with queryset. they are lazy, but you execute them right here. `*_list` authorization methods basically should work with queryset by `.filter()` or `.exclude()`, maybe with `Q()` or `F()`. But not with objects.
regarding business logic - here you need to check if logged in user is not a super user then he should have access only to his `User` object